transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpSubjectSecurityFilter.java

```diff
@@ -357,32 +356,17 @@ void securityMessageReceived(NextFilter nextFilter, IoSession session, Object me
 
         DefaultAuthenticationToken authToken = (DefaultAuthenticationToken) tokenExtractor.extract(httpRequest, realm);
 
-        // If the client request provided authentication data which has
```

Comment:
I see, by moving it you might have fixed an issue for the multiple realm case where subsequent realms don't match the challenge scheme and need a 401??